packages/perennial/contracts/multiinvoker/MultiInvokerRollup.sol

```diff
@@ -267,9 +303,12 @@ contract MultiInvokerRollup is IMultiInvokerRollup, MultiInvoker {
      * @param input 1 byte slice to convert to uint8 to decode lengths
      * @return result The uint8 representation of input
      */
-    function _bytesToUint8(bytes memory input) private pure returns (uint8 result) {
+    function _bytesToUint8(bytes calldata input, uint256 pos) private pure returns (uint8 result) {
         assembly {
-            result := mload(add(input, UINT8_LENGTH))
+            // 1) load calldata into temp starting at ptr position 
+            let temp := calldataload(add(input.offset, pos))
+            // 2) shifts the calldata such that only the first byte (length) is stored in result
+            result := shr(248, temp)
```

Comment:
cleanup idea: it'd be extra gas, but possibly more readable if we did `sub(UINT256_LENGTH, UINT8_LENGTH)` instead of just `248` here

Reply:
its 3 gas lets do it

Reply:
actually we'd have to do shr(ml(8, sub(UINT256_LENGTH, UINT8_LENGTH))) bc its bitlength. id say either add a `constant UINT8_LENGTH_BITS = 8` or just comment on the 248 and keep it

Reply:
yeah I think that's still probably more clear